rllib/algorithms/algorithm.py

```diff
@@ -644,6 +645,17 @@ def setup(self, config: AlgorithmConfig) -> None:
                 )
                 self._evaluation_weights_seq_number = 0
 
+
+        self.evaluation_dataset = None
+        if self.evaluation_config.get("off_policy_estimation_methods"):
+            # the num worker is set to 0 to avoid creating shards since there is no 
+            # worker anymore
+            logger.info("Creating evaluation dataset ...")
+            self.evaluation_dataset, _ = get_dataset_and_shards(
+                self.evaluation_config, num_workers=0
```

Comment:
comment that ``num_workers=0`` is quite important. this will return un-sharded dataset as-is.